airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/transformation.py

```diff
@@ -18,11 +18,11 @@ class RecordTransformation:
     @abstractmethod
     def transform(
         self,
-        record: Record,
+        record: Mapping[str, Any],
         config: Optional[Config] = None,
         stream_state: Optional[StreamState] = None,
         stream_slice: Optional[StreamSlice] = None,
-    ) -> Record:
+    ) -> Mapping[str, Any]:
```

Comment:
I would have updated `AddFields` as well but it meant having to fix interpolation things which brought a lot more MyPy errors so I stopped there